spec/unit/provider/exec/pwsh_spec.rb

```diff
@@ -68,6 +56,28 @@
 
         provider.run_spec_override(command)
       end
+
+      context 'when specifying a path' do
+        let(:path) { Puppet::Util::Platform.windows? ? 'C:/pwsh-test' : '/pwsh-test' }
```

Comment:
So, the problem I was seeing on my machine (and a centos vmpooler machine) is that .[travis.yaml](https://github.com/puppetlabs/puppetlabs-powershell/blob/master/.travis.yml#L10) installs powershell on the SUT. However the machines I was using did not have powershell installed and thus the code at provider/exec/pwsh.rb:19 is throwing "pwsh could not be found".

This dependency should be stubbed out. We should not need to actually install pwsh to complete the unit tests.

Reply:
Right ok.... I've kind of solved this in two parts;

Firstly, those tests require _actually_ calling powershell/pwsh so I've moved those particular tests to `/spec/integration`. This is more admin than fixing though.  I did fix the unit tests to properly stub out any filesystem calls for pwsh so the unit tests will run whether or not pwsh is available.

What I also did is add a guard on those tests to detect if `pwsh` is available and skip them if not.  I'm not 100% happy that I am, but at least it will obvious in the rspec logs e.g.

```
Puppet::Type::Exec::ProviderPwsh
  #run
    returns the output and status (PENDING: Could not locate pwsh binary)
    returns true if the `onlyif` check command succeeds (PENDING: Could not locate pwsh binary)
    returns false if the `unless` check command succeeds (PENDING: Could not locate pwsh binary)
    runs commands properly that output to multiple streams (PENDING: Could not locate pwsh binary)

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Puppet::Type::Exec::ProviderPwsh#run returns the output and status
     # Could not locate pwsh binary
     # ./spec/integration/provider/exec/pwsh_spec.rb:33

  2) Puppet::Type::Exec::ProviderPwsh#run returns true if the `onlyif` check command succeeds
     # Could not locate pwsh binary
     # ./spec/integration/provider/exec/pwsh_spec.rb:41

  3) Puppet::Type::Exec::ProviderPwsh#run returns false if the `unless` check command succeeds
     # Could not locate pwsh binary
     # ./spec/integration/provider/exec/pwsh_spec.rb:48

  4) Puppet::Type::Exec::ProviderPwsh#run runs commands properly that output to multiple streams
     # Could not locate pwsh binary
     # ./spec/integration/provider/exec/pwsh_spec.rb:55
```

Secondly I removed the non-Windows unit tests from the `powershell` provider.  Ideally I wanted to do this as part of MODULES-8359, but after consideration I didn't think I should bother making changes to make the tests pass, to only delete all that effort in 8359 later on.  So instead I just removed the unit tests only.

Reply:
Oh to answer, how did I test this locally, I just renamed `C:\Program Files\PowerShell\6\pwsh.exe` to `pwsh-hidden.exe` so the provider will fail.